.buildkite/publish/dra/init_dra_publishing.sh

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+########
+# publishes a DRA artifact
+########
+
+set -exu
+set -o pipefail
+
+# Load our common environment variables for publishing
+export DRA_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
```

Comment:
sth like that should work better but please test!! 🙏 
```suggestion
export DRA_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
```

Reply:
`realpath` is a function that folks often define in the first lines, like:
```
function realpath {
  echo "$(cd "$(dirname "$1")"; pwd)"/"$(basename "$1")";
}
export BUILDKITE_DIR=$(realpath "$(dirname "$SCRIPT_DIR")")
```

So it's doing the same thing, just as a one-liner (but I think you got dirname/realpath order backwards)

Reply:
I think realpath is [UNIX cmnd](https://man7.org/linux/man-pages/man3/realpath.3.html)
```
(connectors-python) ➜  connectors-python git:(main) ✗ pwd
/Users/jedr/connectors-python
(connectors-python) ➜  connectors-python git:(main) ✗ export DRA_DIR="$(dirname "$(realpath connectors/sources)")"
(connectors-python) ➜  connectors-python git:(main) ✗ echo $DRA_DIR                                               
/Users/jedr/connectors-python/connectors
```

And if I understand correctly you want to get absolute path of the dirname of relative BASH_SOURCE[0] ? In that case I think it should get the order correctly

Nothing blocking though

Reply:
TIL. I've always seen `realpath` defined explicitly. Like: https://github.com/elastic/ent-search/blob/aa06d741001a1e3a5a7a25d7356d1ec49e1a9555/dist/enterprise-search/package_assets/bin/enterprise-search#L8-L10

> Nothing blocking though

In that case, I'm going to leave it as-is, since this script is copied from elsewhere, it's pretty stable/mature. I'd rather make as few changes to it as possible, and avoid letting my own (obvious!) bash inexperience lead to sneaking in a bug.